crates/ra_hir_ty/src/method_resolution.rs

```diff
@@ -271,6 +271,33 @@ pub fn iterate_method_candidates<T>(
     mode: LookupMode,
     mut callback: impl FnMut(&Ty, AssocItemId) -> Option<T>,
 ) -> Option<T> {
+    let mut slot = None;
+    iterate_method_candidates_impl(
+        ty,
+        db,
+        env,
+        krate,
+        traits_in_scope,
+        name,
+        mode,
+        &mut |ty, item| {
+            slot = callback(ty, item);
```

Comment:
also, while we are at it, can `assert!(slot.is_none())` to make sure that impl is correct.